### ISSUE TITLE
ESIPP [Slow] e2e tests flakyness increased

### ISSUE THREAD

**Issue by MrHohn** [kubernetes/kubernetes#56443]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig network
@kubernetes/sig-network-test-failures 

**What happened**:
From:
- http://k8s-testgrid.appspot.com/sig-network-gce#gci-gce-slow&width=5
- https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=to%20have%20no%20load%20balancer

ESIPP e2e tests failed fairly frequently (edit: on waiting for service updates) in CIs since 11/22/2017. Checked the commit diff (https://github.com/kubernetes/kubernetes/compare/490403764...2f6e7bfc0) and seems like https://github.com/kubernetes/kubernetes/pull/55336 may be the cause.

One guess is that after #55336 node sync loop will go in the same work queue as service updates, but given the default concurrency is 1, this might slow down the already slow-moving service update pipeline.

Will spend more time in logs to see if there is any other cause. Example failure:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-slow/12399

**What you expected to happen**:
ESIPP tests should not timeout waiting for LB creation/deletion.

**Anything else we need to know?**:
- Kubernetes version (use `kubectl version`): HEAD (v.1.9+)

cc @jhorwit2 @nicksardo @shyamjvs

**Comment by jhorwit2** [comment]
I've been running that e2e suite (ESIPP.*) the last few days while testing another PR and haven't run into this, but I've been running each test sequentially. Are the tests being ran in parallel? That could definitely slow things down as you mentioned.

**Comment by jhorwit2** [comment]
I only see 2 node changes detected (below)

```
I1127 18:51:48.724263       5 service_controller.go:669] Detected change in list of current cluster nodes. New node set: [bootstrap-e2e-minion-group-3wm6 bootstrap-e2e-minion-group-fh55]
I1127 18:53:28.724569       5 service_controller.go:669] Detected change in list of current cluster nodes. New node set: [bootstrap-e2e-minion-group-3wm6 bootstrap-e2e-minion-group-fh55 bootstrap-e2e-minion-group-wl88]
```

That seems to go fine but then there is a 14 minute gap where there are no syncs. 
```
I1127 19:08:25.241651       5 service_controller.go:761] Service has been deleted e2e-tests-services-xk4vl/mutability-test
I1127 19:08:25.241693       5 service_controller.go:748] Finished syncing service "e2e-tests-services-xk4vl/mutability-test" (63.853µs)
I1127 19:21:00.668053       5 service_controller.go:761] Service has been deleted e2e-tests-esipp-d8n2b/external-local
I1127 19:21:00.668090       5 service_controller.go:748] Finished syncing service "e2e-tests-esipp-d8n2b/external-local" (76.104µs)
```

The timeout was `19:20:41.193`. The deletion log line above was after the timeout and during the cleanup of the test which corresponds with the following logs:
```
STEP: Destroying namespace "e2e-tests-esipp-d8n2b" for this suite.
Nov 27 19:21:01.586: INFO: Waiting up to 30s for server preferred namespaced resources to be successfully discovered
Nov 27 19:21:05.038: INFO: namespace: e2e-tests-esipp-d8n2b, resource: bindings, ignored listing per whitelist
Nov 27 19:21:08.181: INFO: namespace e2e-tests-esipp-d8n2b deletion completed in 12.920987531s
```

As far as events related to the service in question `e2e-tests-esipp-d8n2b/external-local`

```
$ grep service_controller kube-controller-manager.log | grep "e2e-tests-esipp-d8n2b/external-local"
I1127 18:58:31.025599       5 service_controller.go:314] Ensuring LB for service e2e-tests-esipp-d8n2b/external-local
I1127 18:59:22.898597       5 service_controller.go:748] Finished syncing service "e2e-tests-esipp-d8n2b/external-local" (51.873000694s)
I1127 19:04:29.256808       5 service_controller.go:304] Deleting existing load balancer for service e2e-tests-esipp-d8n2b/external-local that no longer needs a load balancer.
E1127 19:05:03.225080       5 service_controller.go:774] Failed to process service e2e-tests-esipp-d8n2b/external-local. Retrying in 5s: googleapi: Error 400: The resource 'projects/k8s-jkns-gce-1-3/global/httpHealthChecks/k8s-8c07463df6cc1ea0-node' is not ready, resourceNotReady
I1127 19:05:03.225092       5 service_controller.go:748] Finished syncing service "e2e-tests-esipp-d8n2b/external-local" (34.30973142s)
I1127 19:06:58.303294       5 service_controller.go:748] Finished syncing service "e2e-tests-esipp-d8n2b/external-local" (3.4µs)
I1127 19:21:00.668053       5 service_controller.go:761] Service has been deleted e2e-tests-esipp-d8n2b/external-local
I1127 19:21:00.668090       5 service_controller.go:748] Finished syncing service "e2e-tests-esipp-d8n2b/external-local" (76.104µs)
```

It's weird the first delete event did not go through the deletion [switch case](https://github.com/oracle/kubernetes/blob/26f9dd7b3e66f10b47163f88eac00ebc33ae0957/pkg/controller/service/service_controller.go#L757). You can tell because it's missing the `Service has been deleted` log line. So far this doesn't seem related to my changes but rather an existing bug.

**Comment by jhorwit2** [comment]
@mrhohn with the help of @liggitt we found the bug causing this. It is unrelated to the pr and existed before. Ill open a pr in a little with the fix.

**Comment by MrHohn** [comment]
@jhorwit2 Thanks for the quick debugging :) Would like to know what is causing this.
